Avoid explicit ssh private keys in autoyast template

RoyCai7 opened the pull request:
Hide keys in some openqa settings, and replace in the autoyast template Progress ticket: 

- Related ticket: https://progress.opensuse.org/issues/160099
- Needles: N/A
- Verification run: 
[15sp5_kvm](http://openqa.qam.suse.cz/tests/overview?build=Test_key_kvm_15-SP5&distri=sle&version=15-SP5&groupid=167)
[15sp5_xen](http://openqa.qam.suse.cz/tests/overview?build=Test_key_xen_15-SP5&distri=sle&version=15-SP5&groupid=168)
[12sp5_kvm](http://openqa.qam.suse.cz/tests/overview?build=Test_key_kvm_12-SP5&distri=sle&version=12-SP5&groupid=149)
[12sp5_xen](http://openqa.qam.suse.cz/tests/overview?build=Test_key_xen_12-SP5&distri=sle&version=12-SP5&groupid=160)
[15sp5_xen_migration](http://openqa.qam.suse.cz/tests/overview?build=Test_key_xen_15-SP5%3Alive_migrat&distri=sle&version=15-SP5&groupid=168)
[15sp5_kvm_all](http://openqa.qam.suse.cz/tests/overview?distri=sle&version=15-SP5&build=Test_key_kvm_15-SP5_all)

alice-suse commented on data/virtualization/autoyast/host_15.xml.ep:
```diff
@@ -358,49 +358,18 @@ cat >> /root/.config/fontconfig/fonts.conf <<EOF
-lbzZvl0rkSQSotKBVWnX/1aU5Pk0RX8HKzYyLZiTjXJAX8RKnem5gsmeVv6yRbcQ2Fp5+I
-t3XODgiOxVK190LwAAAApyb290QHJpdmVyAQIDBAUGBw==
------END OPENSSH PRIVATE KEY-----
+<%= $get_var->('ID_RSA') %>
```
Hi Roy,
Thanks for the quick PR! Super cool!

Just some minor comment:
- for the rsa key, what's the current length? Please use >1024, eg 4096. This is new requirement for 15sp6. 
 You can get your key length by cmd like below
```alice:/home/alice/Documents/github_repo/qa-automation/libs/qa_lib_keys # ssh-keygen -lf qa_lib_keys/id_rsa.pub
4096 SHA256:uZvTO8OP666BqThpABfXwvuPs0uHmBigm8PM2DA6YUI root@qadb2 (RSA)
```

- For setting name of the keys, suggest to have prefix `_SECRET`, which will be hided in openqa jobs( log files, vars...), so more secure. Eg `_SECRET_RSA_PRIV_KEY`, `_SECRET_RSA_PUB_KEY`

RoyCai7 replied:
hansolo:~/.ssh # ssh-keygen -lf id_rsa
3072 SHA256:2Q9+4OlQZAiROq4iJ52ez3DDuJqWYX+cnjYuxLaki1E id_rsa.pub (RSA)
> 1024. it is ok for sles15sp6, I think.

RoyCai7 replied:
Updated setting name of the keys to _SECRET_RSA_PRIV_KEY and _SECRET_RSA_PUB_KEY

alice-suse reviewed (approved):
LGTM

alice-suse commented:
Wait for validation job to finish, then merge if no new comments from others :)

Julie-CAO commented:
LGTM

tbaev reviewed (approved):
LGTM

RoyCai7 commented:
Still have issue for ssh without password. checking.

RoyCai7 commented:
> Still have issue for ssh without password. checking.

Fixed it and updated new code.
Solution:
Convert the contents of id_rsa (which span multiple lines) into a single line by replacing newline characters with 9A.
Then, within the autoyast configuration, use the sed command to replace 9A with newline characters to restore the original format.
After testing, everything works as expected.

alice-suse commented:
@RoyCai7  Are all verifications done? Ready for merge?

RoyCai7 commented:
> @RoyCai7 Are all verifications done? Ready for merge?

Still VR testing.

RoyCai7 commented:
Passed VR test, if no other issue, can be merged.
